backend/src/halo2_impl.rs

```diff
@@ -16,8 +16,19 @@ impl<F: FieldElement> BackendFactory<F> for Halo2ProverFactory {
         _output_dir: Option<&'a Path>,
         setup: Option<&mut dyn io::Read>,
         verification_key: Option<&mut dyn io::Read>,
+        options: BackendOptions,
     ) -> Result<Box<dyn crate::Backend<'a, F> + 'a>, Error> {
-        let mut halo2 = Box::new(Halo2Prover::new(pil, fixed, setup)?);
+        let proof_type = match options.as_str() {
+            "" | "poseidon" => ProofType::Poseidon,
+            "snark_single" => ProofType::SnarkSingle,
+            "snark_aggr" => ProofType::SnarkAggr,
+            _ => {
+                return Err(Error::BackendError(format!(
+                    "Invalid proof type: {options}"
+                )))
+            }
+        };
```

Comment:
Implemented this for Halo2 as well now. One annoying thing is that since Halo2 is a separate crate we don't have access to the BackendOptions type. Used String for now. I think we should move Halo2 from a crate to a module inside the `backend` crate.